Fix for ability to activate buttons using right click

Comandeer opened the pull request:
## What is the purpose of this pull request?

Bug fix

## Does your PR contain necessary tests?

All patches which change the editor code must include tests. You can always read more
on [PR testing](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_contributing_code.html#tests),
[how to set the testing environment](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html) and
[how to create tests](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html#creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [x] Unit tests
- [x] Manual tests

## What changes did you make?

I've added additional check if left mouse button was used for click. I also add ability to get info about mouse button from native DOM event in `CKEDITOR.tools.getMouseButton`.

Closes #2565.

Comandeer commented on tests/plugins/button/button.js:
```diff
@@ -41,5 +52,45 @@ bender.test( {
+	var ie8ButtonMap = {
+			0: 1,
+			1: 4,
+			2: 2
```
Comments sound like a good idea 👍

Comandeer commented on core/tools.js:
```diff
@@ -1464,17 +1464,18 @@
-				// Added in case when there's no data available. That's the case in some unit test in built version which
-				// mock event but doesn't put data object.
+			if ( evt instanceof Event ) {
+				domEvent = evt;
```
I'd rather keep it the way it is. After inlining we'd have two conditionals (ternary operator and `&&`) in one line.

Comandeer commented:
The issue with Edge is present also on current `master` branch, so it's a separate bug. What's more you must click toolbar button as the first thing after selecting text, clicking anywhere else first and then on button does not trigger the bug. However I don't think there is much sense in reporting it, as it's a real edge case (no pun intended).

engineering-this commented:
I could resolve my own suggestion, but only the first one. For the second GH said 'You are not allowed to resolve suggestions'. So I just manually pushed commit 😂

Comandeer commented:
Fixed failing tests. It was connected with too strict conditional checking if the event is instance of native DOM event. Now the function assumes that passed `evt` option is native DOM event if it doesn't have `data` property.

I've also refactored button's template according to the advice to be more readable.
